assets/js/http.js

```diff
@@ -11,4 +11,52 @@ class Http {
 
     return data;
   }
+
+  async post(url, data) {
+    const response = await fetch(url, {
+      method: "POST",
+      headers: {
+        "Content-type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+
+    const resData = await response.json();
+
+    return resData;
+  }
+
+  async put(url, data) {
+    const response = await fetch(url, {
+      method: "PUT",
+      headers: {
+        "Content-type": "application/json",
+      },
+      body: JSON.stringify(data),
+    });
+
+    if (!response.ok) {
+      const message = `An error has occured: ${response.status}`;
+      throw new Error(message);
+    }
+
+    const resData = await response.json();
+
+    return resData;
+  }
+
+  async delete(url) {
+    const response = await fetch(url, {
+      method: "DELETE",
+    });
+
+    if (!response.ok) {
+      const message = `An error has occured: ${response.status}`;
+      throw new Error(message);
+    }
+
+    const resData = "Product deleted";
```

Comment:
Nitpick, this looks like a common http/fetch wrapper functions, so, it would be best to avoid custom messages related to some domain.  
You could also just return a `true` as response.  
Or even the response it self? 😃